File: app/javascript/channels/game_channel.js
import consumer from "./consumer";

// Usage: call subscribeToGameChannel(gameId, callbacks)
export function subscribeToGameChannel(
  gameId,
  { received, connected, disconnected } = {}
) {
  return consumer.subscriptions.create(
    { channel: "GameChannel", game_id: gameId },
    {
      connected() {
        if (connected) connected();
      },
      disconnected() {
        if (disconnected) disconnected();
      },
      received(data) {
        if (received) received(data);
      },
    }
  );
}

File: app/channels/game_channel.rb
class GameChannel < ApplicationCable::Channel
  def subscribed
    # Stream from a game-specific stream (to be customized per game instance)
    stream_from "game_#{params[:game_id]}"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def receive(data)
    # Handle data received from the client (e.g., moves, chat)
    # ActionCable.server.broadcast("game_", data) can be used for rebroadcasting
  end
end 